sourcegraph/sourcegraph
# Add a screen to check if user is logged in with the same account on web as cody client

Context: https://sourcegraph.slack.com/archives/C05AGQYD528/p1713500117867789?thread_ts=1713416841.706729&cid=C05AGQYD528

VS Code PR: https://github.com/sourcegraph/cody/pull/3868

E2E Demo: https://www.loom.com/share/7b525676583b4ad6aa38972a8138f4cd?sid=a24403de-8c0f-4bba-b343-2f9214651402

The VS Code PR adds a query parameter `cody_client_user=<username>` while redirecting to `/cody/manage`. 

In this PR, we check for the query param on the web and verify if the user is logged in with the same account. If not, we redirect them to the switch account screen. The screen gives them a disclaimer and option to either sign out and switch to the correct account or to continue with the one they are logged in right now.

Screenshot:
![image](https://github.com/sourcegraph/sourcegraph/assets/22571395/71827602-525f-479b-afb3-449c29afe68c)

## Test plan
- checkout the vs code [PR](https://github.com/sourcegraph/cody/pull/3868) locally, build the extension and connect to the local sg instance
- signing to account A in the VS Code extension.
- got to sourcegraph.test:3443, signout and sign in into account B.
- Go to Cody in VS Code, click on account -> manage account -> redirect
- You should be redirected to the switch account page. 
- Test both continue and signout actions. 
- After signout on web, signin with account A. 
- Go to Cody in VS Code again, click on account -> manage account -> redirect
- This time you should land at `/cody/manage` and the switch account screen should not be visible.
